### ISSUE TITLE
N1 quits unexpectedly after closing main window and reopening using OSX Dock icon

### ISSUE THREAD

**Issue by paulicelli** [nylas/N1#481]
Quits unexpectedly after reopening the main window

Steps to reproduce:
1. N1 is closed
2. open N1 using the dock icon
3. close the main window using the red button (N1 is still open in the dock, as indicated by the small dot)
4. click on the dock icon (with the intent to reopen the main window)
5. N1 quits unexpectedly

This is not occurring if you change the point 3 with:
3. hide the main window using the shortcut cmd+h

Version 0.3.25-c174641 (0.3.25-c174641)

**Comment by akizor** [comment]
Same thing here. Quits unexpectedly when reopening.
OSX: 10.11
N1: Version 0.3.25-c174641 (0.3.25-c174641)

**Comment by scubakyle** [comment]
The same thing happens for me, even when using cmd+w to close the window.

OSX 10.11.1
N1 Version 0.3.25-c174641 (0.3.25-c174641)

**Comment by Muurtegel** [comment]
I have the same thing. I'm not sure if this helps, but this is what I get in the console:

```
26/11/15 17:03:59,461 Nylas[28843]: TypeError: Cannot read property 'preventDefault' of undefined
    at EventEmitter.<anonymous> (/Applications/Nylas N1.app/Contents/Resources/app.asar/src/browser/application.js:509:23)
    at emitNone (events.js:67:13)
    at EventEmitter.emit (events.js:166:7)
    at EventEmitter.<anonymous> (/Applications/Nylas N1.app/Contents/Resources/atom.asar/browser/api/lib/app.js:99:19)
    at EventEmitter.<anonymous> (/Applications/Nylas N1.app/Contents/Resources/atom.asar/common/api/lib/deprecate.js:73:21)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)
```

**Comment by mbilker** [comment]
@Muurtegel Ok. Thank you. The method called is:
```
app.on('activate-with-no-open-windows', (function(_this) {
        return function(event) {
          _this.openWindowsForTokenState();
          return event.preventDefault();
        };
      })(this));
```

**Comment by jstejada** [comment]
hey guys this was fixed with #497 , so I'm closing this
